{
  "instructions": [
    "let parseAssertElemProp = await page.$(\"a\");\nif (parseAssertElemProp === null) { throw '\"a\" not found'; }\nawait page.evaluate(e => {\n    const nonMatchingProps = [];\n    const parseAssertElemPropDict = {\"a\":\"1\"};\n    for (const [parseAssertElemPropKey, parseAssertElemPropValue] of Object.entries(parseAssertElemPropDict)) {\n        if (e[parseAssertElemPropKey] === undefined) {\n            continue;\n        }\n        if (String(e[parseAssertElemPropKey]) == parseAssertElemPropValue) {\n            nonMatchingProps.push(\"assert didn't fail for property `\" + parseAssertElemPropKey + '`');\n        }\n    }\n    if (nonMatchingProps.length !== 0) {\n        const props = nonMatchingProps.join(\", \");\n        throw \"The following errors happened (for selector `a`): [\" + props + \"]\";\n    }\n}, parseAssertElemProp);"
  ],
  "wait": false,
  "warnings": [
    "Pseudo-elements (`::after`) don't have attributes so the check will be performed on the element itself"
  ],
  "checkResult": true
}
